core/server/master/src/main/java/alluxio/master/file/meta/TtlBucketList.java

```diff
@@ -89,8 +89,19 @@ public void insert(Inode<?> inode) {
       // start time of this interval should be (ttlEndTimeMs / interval) * interval.
       long interval = TtlBucket.getTtlIntervalMs();
       bucket = new TtlBucket(interval == 0 ? ttlEndTimeMs : ttlEndTimeMs / interval * interval);
-      mBucketList.add(bucket);
+      synchronized (mBucketList) {
```

Comment:
I see, good catch indeed. I would prefer to avoid locking and instead use optimistic concurrency:

```
    TtlBucket bucket;
    while (true) {
      bucket = getBucketContaining(inode);
      if (bucket != null) {
        break;
      }
      long ttlEndTimeMs = inode.getCreationTimeMs() + inode.getTtl();
      ...
      long interval = TtlBucket.getTtlIntervalMs();
      bucket = new TtlBucket(interval == 0 ? ttlEndTimeMs : ttlEndTimeMs / interval * interval);
      if (mBucketList.add(bucket)) {
        break;
      }
    }
    bucket.addInode(inode);
```